### PR TITLE
13702 contents api service

### PR DESCRIPTION
**Pull request by burzomir**

### CODE REVIEW

File: src/common/api/order-builder.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Copyright (C) The Arvados Authors. All rights reserved.
+//
+// SPDX-License-Identifier: AGPL-3.0
+
+
+export default class OrderBuilder {
+    private order: string[] = [];
```
Comment (janicka1) [inline src/common/api/order-builder.ts]: `private _order` ?

File: src/common/api/common-resource-service.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                            [newKey]: CommonResourceService.mapKeys(mapFn)(value[key])
+                        }), {});
+                case _.isArray(value):
+                    return value.map(CommonResourceService.mapKeys(mapFn));
+                default:
+                    return value;
+            }
+        }
+
+    constructor(serverApi: AxiosInstance, resourceType: string) {
```
Comment (janicka1) [inline src/common/api/common-resource-service.ts]: `constructor(private _serverApi: AxiosInstance, resourceType: string) {`

File: src/common/api/common-resource-service.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    kind: string;
+    offset: number;
+    limit: number;
+    items: T[];
+    itemsAvailable: number;
+}
+
+export default class CommonResourceService<T extends Resource> {
+
+    serverApi: AxiosInstance;
```
Comment (janicka1) [inline src/common/api/common-resource-service.ts]: private methods

File: src/common/api/common-resource-service.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                case _.isArray(value):
+                    return value.map(CommonResourceService.mapKeys(mapFn));
+                default:
+                    return value;
+            }
+        }
+
+    constructor(serverApi: AxiosInstance, resourceType: string) {
+        this.serverApi = serverApi;
+        this.resourceType = "/" + resourceType;
```
Comment (janicka1) [inline src/common/api/common-resource-service.ts]: this.resourceType = "/" + resourceType + '/'

File: src/common/api/order-builder.ts (hunk 1 above)
Comment (burzomir) [reply src/common/api/order-builder.ts]: Current tslint config does not allow that.

**Pull request merged**